engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/transformation/DefinedParameter.scala

```diff
@@ -1,14 +1,29 @@
 package pl.touk.nussknacker.engine.api.context.transformation
 
-import pl.touk.nussknacker.engine.api.typed.typing.{TypingResult, Unknown}
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypingResult, Unknown}
 
 sealed trait DefinedParameter {
   def returnType: TypingResult
 }
 
-case class DefinedLazyParameter(returnType: TypingResult) extends DefinedParameter
-case class DefinedEagerParameter(value: Any, returnType: TypingResult) extends DefinedParameter
-case object FailedToDefineParameter extends DefinedParameter {
-  override def returnType: TypingResult = Unknown
+sealed trait DefinedSingleParameter extends DefinedParameter
+
+sealed trait DefinedBranchParameter extends DefinedParameter {
+
+  override def returnType: TypingResult = Typed(detailedReturnType.values.toSet)
+
+  def detailedReturnType: Map[String, TypingResult]
+
+}
+
+case class DefinedLazyParameter(returnType: TypingResult) extends DefinedSingleParameter
+case class DefinedEagerParameter(value: Any, returnType: TypingResult) extends DefinedSingleParameter
+
+case class DefinedLazyBranchParameter(detailedReturnType: Map[String, TypingResult]  ) extends DefinedBranchParameter
```

Comment:
Why it is `detailed` and for single version is not? More benefits would be from adding `ByBranchId` suffix because someone can think that this `String` is for e.g. some variable name that is common usage when you see map string -> type ...

Reply:
:+1: I'll add it